Shift3/boilerplate-client-react
# Added `Delete User` functionality

Handles #305

## Changes
1. Added `DELETE` endpoint in `userApi`.
2. Changed directory name to `user-dashboard`.
3. Added provides/invalidates tags to api endpoints to automatically refetch cashed data after user is deleted. 
4. Modified `onClick` handler of the `Delete` action button in the user table to open up a confirmation modal to delete the user.
5. Routed the user content  to the `/users` path instead of `/admin` path.

## Testing
1. Pull in the changes to your local copy of this branch and run `yarn start`.
2. Log in and click delete user.
3. Log out and back in and check that the user is deleted.

### Closes #305
